packages/design-system-react/src/components/button-base/ButtonBase.tsx

```diff
@@ -0,0 +1,99 @@
+import { Slot } from '@radix-ui/react-slot';
+import React from 'react';
+
+import { twMerge } from '../../utils/tw-merge';
+import { Icon } from '../icon';
+import { SIZE_CLASS_MAP } from './ButtonBase.constants';
+import type { ButtonBaseProps } from './ButtonBase.types';
+import { ButtonBaseSize } from './ButtonBase.types';
+
+export const ButtonBase = React.forwardRef<HTMLButtonElement, ButtonBaseProps>(
+  (
+    {
+      children,
+      className,
+      size = ButtonBaseSize.Md,
+      isFullWidth,
+      asChild,
+      isDisabled,
+      isLoading,
+      loadingText,
+      startIconName,
+      startIconProps,
+      startAccessory,
+      endIconName,
+      endIconProps,
+      endAccessory,
+      ...props
+    },
+    ref,
+  ) => {
+    const Component = asChild ? Slot : 'button';
+
+    const mergedClassName = twMerge(
+      // Base styles
+      'inline-flex items-center justify-center',
+      'rounded-full px-4',
+      'text-default font-medium',
+      'transition-colors duration-150',
+      'bg-muted hover:bg-muted-hover active:bg-muted-pressed',
+      // Size
+      SIZE_CLASS_MAP[size],
+      // Full width
+      isFullWidth && 'w-full',
+      // Disabled state
+      isDisabled && 'opacity-50 cursor-not-allowed',
+      // Loading state
+      isLoading && 'cursor-wait',
+      // Custom classes
+      className,
+    );
```

Comment:
I've created an issue to improve the way we apply tailwind classnames to components and possibly adopting the [class variance authority](https://cva.style/docs) approach which may align better with how engineers that like CSS in JS use Tailwind